### PR TITLE
Merge evenstream generation fixes

### PR DESCRIPTION
**Pull request by dagnir**
## Description
Merge in the event stream code generation fixes from `dongie/eventstream-visitors-fix` branch.

## Motivation and Context
Bug fix.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Review by Quanzzzz — APPROVED** [review]

File: codegen/src/test/java/software/amazon/awssdk/codegen/poet/eventstream/EventModelSpecTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package software.amazon.awssdk.codegen.poet.eventstream;
```
[the remark this answers is not in the document]
Comment (dagnir) [reply codegen/src/test/java/software/amazon/awssdk/codegen/poet/eventstream/EventModelSpecTest.java:1]: Yep, good catch. I'll add it

**Comment by dagnir** [comment]
Diff of generated models for Kinesis and Transcribe Streaming before and after the generation changes: https://gist.github.com/dagnir/f26631daf0ed8a2b2a595ac73357d68f

**Comment by zoewangg** [comment]
`@SafeVarargs` got removed somehow

https://gist.github.com/dagnir/f26631daf0ed8a2b2a595ac73357d68f#file-kinesis-diff-L5648

**Comment by zoewangg** [comment]
Can we add a changelog entry?

**Comment by dagnir** [comment]
> @SafeVarargs got removed somehow

Good change, this was a change that I ended up not needing. Will revert.

> Can we add a changelog entry?

Sure

**Review by zoewangg — APPROVED** [review]

**Review by Quanzzzz — APPROVED** [review]

**Pull request merged**